packages/core/src/components/tabs/tabs.tsx

```diff
@@ -0,0 +1,157 @@
+      const currentActiveTab = this.tabs[currentActive];
+      const nextActiveTab = this.tabs[nextActive];
+
+      if (currentActiveTab && nextActiveTab) {
+        currentActiveTab.active = false;
+        nextActiveTab.active = true;
+        this.activeTab = tabId;
+        this.host.shadowRoot?.querySelector<HTMLAnchorElement>(`#${tabId} > a`)?.focus();
+      }
+    }
+  }
+
+  componentWillLoad(): void | Promise<void> {
+    const tabElements = this.host.querySelectorAll("dso-tab");
+
+    if (tabElements) {
+      this.tabs = Array.from(tabElements)
+        .map<Tab | undefined>((tab) => {
+          const id = tab.getAttribute("identifier");
+
+          if (id) {
+            return {
+              label: tab.label,
+              id,
+              active: tab.active,
+              disabled: tab.disabled,
+            };
+          }
+
+          return undefined;
+        })
+        .filter<Tab>((tab): tab is Tab => !!tab);
+    }
+
+    this.activeTab = this.tabs?.find((tab) => tab.active)?.id;
+  }
+
+  private emitEvent = (originalEvent: MouseEvent | KeyboardEvent, selected: string) => {
+    this.dsoTabSwitch.emit({
+      originalEvent,
+      selected,
+    });
+  };
+
+  private keyUpHandler = (event: KeyboardEvent) => {
+    const anchor = event.currentTarget;
+    if (
+      !(event.key === "ArrowLeft" || event.key === "ArrowRight") ||
+      !(anchor instanceof HTMLAnchorElement) ||
+      !this.tabs
+    ) {
+      return;
+    }
+
+    const currentTab = this.tabs.find((tab) => tab.id === anchor.closest("li")?.id);
+
+    if (currentTab) {
+      const enabledTabs = this.tabs?.filter((tab) => !tab.disabled);
+      const indexOfCurrentTab = enabledTabs.indexOf(currentTab);
+
+      if (typeof indexOfCurrentTab === "number") {
+        let selected;
+
+        if (event.key === "ArrowLeft" && indexOfCurrentTab > 0) {
+          selected = enabledTabs[indexOfCurrentTab - 1]?.id;
+        }
+
+        if (event.key === "ArrowRight" && indexOfCurrentTab < this.tabs.length - 1) {
+          selected = enabledTabs[indexOfCurrentTab + 1]?.id;
+        }
+
+        if (selected) {
+          this.emitEvent(event, selected);
+        }
+      }
+    }
+  };
+
+  private clickHandler = (event: MouseEvent) => {
+    if (!this.tabs) {
+      return;
+    }
+
+    event.preventDefault();
+
+    const anchor = event.currentTarget;
+
+    if (!(anchor instanceof HTMLAnchorElement)) {
+      return;
+    }
+
+    const selected = anchor.closest("li")?.id;
+    const currentTab = this.tabs.find((tab) => tab.id === selected);
+
+    if (selected && currentTab && !currentTab.disabled) {
+      this.emitEvent(event, selected);
+    }
+  };
+
+  render() {
+    return (
+      <>
+        <ul class="nav nav-tabs" role="tablist">
+          {this.tabs?.map((tab) => (
+            <li role="presentation" class={clsx({ active: tab.active, disabled: tab.disabled })} id={tab.id}>
+              <a
+                href="#"
```

Comment:
de href target is altijd de id van de bijbehorende tabpanel. een werkende link dus, ook zonder js.

Reply:
Ik begrijp wat je bedoelt, maar ik ben er niet heel gevoelig voor omdat het hele stelsel niet zonder JavaScript werkt.

Hoe verhoudt de URL van de tab zich tot de adresbalk en de router van de applicatie?

Reply:
Dit hebben we anders opgelost, we gaan zowel `<button>` als `<a>` ondersteunen